### PR TITLE
allow override checking to proceed past obsolete check

### PR DESCRIPTION
**Pull request by TyOverby**
closes https://github.com/dotnet/roslyn/issues/11691

### CODE REVIEW

**Comment by TyOverby** [comment]
please review @dotnet/roslyn-compiler

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs [hunk 1]
```diff
@@ -2997,5 +2997,35 @@ public static void Main()
...
+using System;
+public abstract class BaseClass<T>
+{
+    public abstract int Method(T input);
+}
+
+public class DerivingClass<T> : BaseClass<T>
+{
+    [Obsolete(""Deprecated"")]
+    public override void Method(T input)
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs]: Have we checked to see if VB has the same bug?  Even if not we should add a regression suite.

**Comment by TyOverby** [comment]
@dotnet-bot test windows_debug_unit64_prtest please

**Comment by TyOverby** [comment]
@dotnet-bot test windows_debug_unit64_prtest please

**Comment by TyOverby** [comment]
VB does not exhibit this bug.  A regression test for VB has been added nonetheless.

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb [hunk 2]
```diff
@@ -2946,5 +2946,35 @@ BC37256: An error occurred while writing the output file: <%= output.ThrownExcep
...
+        Public Overrides Sub Method(input As T)
+        Throw New NotImplementedException()
+    End Sub
+End Class
+]]>
+                    </file>
+                </compilation>)
+
+            compilation.VerifyDiagnostics(
+                Diagnostic(ERRID.ERR_ExpectedEOS, "Inherits").WithLocation(7, 34),
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: The source has a parse error.

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb [hunk 3]
```diff
@@ -2946,5 +2946,34 @@ BC37256: An error occurred while writing the output file: <%= output.ThrownExcep
...
+                <compilation>
+                    <file name="a.vb">
+                        <![CDATA[
+imports System
+
+Public MustInherit Class BaseClass(of T)
+    Public MustOverride Function Method(input As T) As Integer
+End Class
+    
+Public Class DerivingClass(Of T) Inherits BaseClass(Of T)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: >Inherits [](start = 33, length = 8)

Please add a line break before Inherits.

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb [hunk 4]
```diff
@@ -2946,5 +2946,34 @@ BC37256: An error occurred while writing the output file: <%= output.ThrownExcep
...
+    <Obsolete("Deprecated")>
+    Public Overrides Sub Method(input As T)
+        Throw New NotImplementedException()
+    End Sub
+End Class
+]]>
+                    </file>
+                </compilation>)
+
+            compilation.VerifyDiagnostics(
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: >VerifyDiagnostics [](start = 24, length = 17)

Please use AssertTherseDiagnostics helper instead, see previous unit-test for an example.

**Comment by AlekseyTs** [comment]
The new VB unit-test requires some fix-ups as indicated by comments above, LGTM otherwise.

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb [hunk 5]
```diff
@@ -2946,5 +2946,34 @@ BC37256: An error occurred while writing the output file: <%= output.ThrownExcep
 </expected>)
             End Using
         End Sub
+
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: Nit: double blank line.

Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: Done

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb (hunk 3 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: Done

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb (hunk 4 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: Done.

File: src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb (hunk 2 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/Emit/CompilationEmitTests.vb]: Oops, fixed

**Comment by AlekseyTs** [comment]
LGTM

**Comment by TyOverby** [comment]
@dotnet-bot test windows_release_unit64_prtest please

**Pull request merged**